### PR TITLE
fix removal of torrent files with a folder structure

### PR DESCRIPTION
**Pull request by sweetppro**
fix for https://github.com/transmission/transmission/issues/3085 and possibly https://github.com/transmission/transmission/issues/2379

### CODE REVIEW

File: libtransmission/torrent-files.cc [hunk 1]
```diff
@@ -218,13 +218,10 @@ bool tr_torrent_files::move(
     {
         auto const remove_empty_directories = [](char const* filename)
         {
-            if (isEmptyDirectory(filename))
```
Comment (ile6695) [inline libtransmission/torrent-files.cc:221]: Will `remove_empty_directories` now remove directories that aren't empty?

Comment (sweetppro) [reply libtransmission/torrent-files.cc:221]: @ile6695 Tbh I’m not sure…
In what situation is it desirable to keep non empty folders. why would you ever need to keep some files in the original location?

Comment (sweetppro) [reply libtransmission/torrent-files.cc:221]: hmm, ok.
ill have a play around tonight

Comment (sweetppro) [reply libtransmission/torrent-files.cc]: @ckerr @ile6695 ive re-added,
the fix still works with the **isEmptyDirectory** check

**Comment by ckerr** [comment]
TBH I don't really understand either this bug or this fix. e.g. where is the `unfinished/` folder mentioned in 3085 coming from?

Would it be possible to add something to `tests/libtransmission/remove-test` to demonstrate this bug so that I can better understand it?

File: libtransmission/torrent-files.cc [hunk 2]
```diff
@@ -267,7 +267,8 @@ void tr_torrent_files::remove(std::string_view parent_in, std::string_view tmpdi
 
     // Make a list of the top-level torrent files & folders
     // because we'll need it below in the 'remove junk' phase
-    auto top_files = std::set<std::string>{};
+    auto const path = tr_pathbuf{ parent, '/', tmpdir_prefix };
+    auto top_files = std::set<std::string>{ std::string{ path } };
```
Comment (ckerr) [inline libtransmission/torrent-files.cc:271]: What files are being left over that requires this? And why does the current code not remove them...?

Comment (GaryElshaw) [reply libtransmission/torrent-files.cc:271]: I've seen this when downloading and then adding a new torrent where the source files are the same. You remove the old torrent, but the initially downloaded files remain.

Comment (sweetppro) [reply libtransmission/torrent-files.cc:271]: @ckerr on macOS Transmission downloads to a temporary location:
`~/Library/Application\ Support/Transmission/unfinished`

when the download is complete, it is moved to your default location, which in my case is my `Downloads` folder:
Screen Shot 2022-05-31 at 6 22 45 am

the bug occurs when you download anything which contains a folder, eg:
https://archive.org/details/parmenides-poem-balboa

the torrent downloads just fine, and is moved to my default location when downloading finishes just fine. But the containing folder `parmenides-poem-balboa` is not removed from my `unfinished folder`

**Pull request merged**